pkg/reconciler/taskrun/resources/taskref.go

```diff
@@ -124,7 +124,21 @@ func GetTaskFunc(ctx context.Context, k8s kubernetes.Interface, tekton clientset
 }
 
 // GetStepActionFunc is a factory function that will use the given Ref as context to return a valid GetStepAction function.
-func GetStepActionFunc(tekton clientset.Interface, namespace string) GetStepAction {
+// also requires a kubeclient, tektonclient, requester in case it needs to find that task in
+// cluster or authorize against an external repositroy. It will figure out whether it needs to look in the cluster or in
+// a remote image to fetch the reference.
```

Comment:
may be mistaken, but i'd expect this to be "location" not "image"
```suggestion
// a remote location to fetch the reference.
```

Reply:
Yeah, we should use location, the `image` refers to bundle image,